core/lib/dal/src/applied_snapshot_status_dal.rs

```diff
@@ -0,0 +1,100 @@
+use zksync_types::{snapshots::AppliedSnapshotStatus, L1BatchNumber};
+
+use crate::StorageProcessor;
+
+#[derive(Debug)]
+pub struct AppliedSnapshotStatusDal<'a, 'c> {
+    pub(crate) storage: &'a mut StorageProcessor<'c>,
+}
+
+impl AppliedSnapshotStatusDal<'_, '_> {
+    pub async fn set_applied_snapshot_status(
+        &mut self,
+        status: &AppliedSnapshotStatus,
+    ) -> Result<(), sqlx::Error> {
```

Comment:
Nit: You can use `sqlx::Result<()>` here; it's used in most other cases.

Reply:
corrected